### PR TITLE
Remove legacy gossip

### PR DESCRIPTION
**Pull request by alarso16**
## Why this should be merged
This removes the deprecated legacy gossip manager.

## How this works
All revisions exactly follow the recommendations laid out in #1428, following the example set in the [coreth PR](https://github.com/ava-labs/subnet-evm/issues/1428). Additionally, a change to `plugin/evm/vm.go` to completely remove the implementation of the deprecated handlers.

## How this was tested
Unit tests

## Need to be documented?
No

## Need to update RELEASES.md?
Yes - updated.

### CODE REVIEW

File: plugin/evm/message/codec.go [hunk 1]
```diff
@@ -15,19 +15,14 @@ const (
...
+var Codec codec.Manager
 
 func init() {
 	Codec = codec.NewManager(maxMessageSize)
 	c := linearcodec.NewDefault()
 
 	errs := wrappers.Errs{}
 	errs.Add(
-		// Gossip types
-		c.RegisterType(EthTxsGossip{}),
```
Comment (darioush) [inline plugin/evm/message/codec.go:29]: We should replace this with:
`c.SkipRegistrations(1)`
this will keep the numerical types registered below unchanged.

Comment (alarso16) [reply plugin/evm/message/codec.go:29]: Added above err wrapper.

File: peer/network_test.go [hunk 2]
```diff
@@ -503,15 +505,15 @@ func TestHandleInvalidMessages(t *testing.T) {
...
 	clientNetwork.SetRequestHandler(&testRequestHandler{})
 
 	assert.NoError(t, clientNetwork.Connected(context.Background(), nodeID, defaultPeerVersion))
 
 	defer clientNetwork.Shutdown()
 
 	// Ensure a valid gossip message sent as any App specific message type does not trigger a fatal error
-	gossipMsg, err := buildGossip(codecManager, HelloGossip{Msg: "hello there!"})
+	marshaller := helloGossipMarshaller{codec: codecManager}
+	gossipMsg, err := marshaller.MarshalGossip(&HelloGossip{Msg: "hello sthere!"})
```
Comment (qdm12) [inline peer/network_test.go:516]: nit there is an extra s! 😨 

```suggestion
	gossipMsg, err := marshaller.MarshalGossip(&HelloGossip{Msg: "hello there!"})
```

**Review by darioush — DISMISSED** [review]

File: RELEASES.md [hunk 3]
```diff
@@ -3,6 +3,7 @@
 ## Pending Release
 
 * Refactored trie_prefetcher.go to be structurally similar to upstream.
+* Remove legacy gossip handler and metrics (deprecated since v0.10.0)
```
Comment (ceyonur) [inline RELEASES.md:6]: coreth and subnet-evm versions are different. Can we just write "removed deprecated legacy gossip handler and metrics"?

File: peer/network_test.go [hunk 4]
```diff
@@ -719,28 +716,22 @@ type HelloGossip struct {
...
 }
 
-func (t *testGossipHandler) HandleEthTxs(nodeID ids.NodeID, msg message.EthTxsGossip) error {
-	t.received = true
-	t.nodeID = nodeID
-	return nil
+func (g helloGossipMarshaller) UnmarshalGossip(bytes []byte) (*HelloGossip, error) {
+	h := &HelloGossip{}
+	_, error := g.codec.Unmarshal(bytes, h)
+	return h, error
```
Comment (qdm12) [inline peer/network_test.go:734]: nit convention in Go (🤷) is to have error variables as `err`

```suggestion
	_, err := g.codec.Unmarshal(bytes, h)
	return h, err
```

File: peer/network_test.go [hunk 5]
```diff
@@ -30,6 +30,10 @@ import (
 	"github.com/ava-labs/avalanchego/version"
 )
 
+const (
+	codecVersion uint16 = 0
+)
```
Comment (qdm12) [inline peer/network_test.go:35]: nit: since this only used for tests, perhaps let's rename it to `testCodecVersion` to avoid any confusion?

Also would it be possible to use another version, like `1`? Personal opinion, it's nice to test with non-zero values to verify nothing was forgotten (since default = zero value)

File: RELEASES.md (hunk 3 above)
Comment (alarso16) [reply RELEASES.md]: Fixed

**Review by ceyonur — APPROVED** [review]

File: plugin/evm/message/codec.go [hunk 6]
```diff
@@ -15,19 +15,17 @@ const (
 	maxMessageSize = 2*units.MiB - 64*units.KiB // Subtract 64 KiB from p2p network cap to leave room for encoding overhead from AvalancheGo
 )
 
-var (
-	Codec codec.Manager
-)
+var Codec codec.Manager
```
Comment (qdm12) [inline plugin/evm/message/codec.go:18]: nit unneeded diff, let's revert it

```suggestion
var (
	Codec codec.Manager
)
```

File: plugin/evm/message/handler.go [hunk 7]
```diff
@@ -6,27 +6,10 @@ package message
...
-	HandleEthTxs(nodeID ids.NodeID, msg EthTxsGossip) error
-}
-
-type NoopMempoolGossipHandler struct{}
-
-func (NoopMempoolGossipHandler) HandleEthTxs(nodeID ids.NodeID, msg EthTxsGossip) error {
-	log.Debug("dropping unexpected EthTxsGossip message", "peerID", nodeID)
-	return nil
-}
+var _ RequestHandler = NoopRequestHandler{}
```
Comment (qdm12) [inline plugin/evm/message/handler.go:12]: nit to reduce git diffs

```suggestion
var (
	_ RequestHandler = NoopRequestHandler{}
)
```

**Review by qdm12 — APPROVED** [review]
I like 🔴  deletions 👍 
Just a few nitpicky comments here and there, but looks good 💯

**Review by ceyonur — APPROVED** [review]

**Review by darioush — APPROVED** [review]
